x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/type/DataTypeConversion.java

```diff
@@ -145,6 +161,8 @@ private static Conversion conversion(DataType from, DataType to) {
                 return conversionToFloat(from);
             case DOUBLE:
                 return conversionToDouble(from);
+            case DATE:
```

Comment:
Please be consistent - either DATE comes before DATETIME or vice-versa - in the Conversion check above it's the other way around.